test/Response/SapiStreamEmitterTest.php

```diff
@@ -57,76 +61,322 @@ public function testDoesNotInjectContentLengthHeaderIfStreamSizeIsUnknown()
         }
     }
 
-    public function emitBodyProvider()
+    private function expandTestData()
```

Comment:
While I get that you wanted to normalize the mocking done for the data providers, this method has become absolutely unreadable. I would suggest duplicating or the supported scenarios. Specifically based on:

 * number of passed in parameters
 * types of passed in parameters